policy/release/olm.rego

```diff
@@ -122,6 +122,34 @@ deny contains result if {
 	result := lib.result_helper(rego.metadata.chain(), [error])
 }
 
+# METADATA
+# title: Unpinned images in input snapshot
+# description: >-
+#   Check the input snapshot for the presence of unpinned image references.
+#   Unpinned image pull references are references to images
+#   that do not contain a digest -- uniquely identifying the version of
+#   the image being pulled.
+# custom:
+#   short_name: unpinned_snapshot_references
+#   failure_msg: The %q image reference is not pinned in the input snapshot.
+#   solution: >-
+#     Update the input snapshot replacing the unpinned image reference with pinned image
+#     reference. Pinned image reference contains the image digest.
+#   collections:
+#   - redhat
+#   effective_on: 2024-08-01T00:00:00Z
+#
+deny contains result if {
+	_release_restrictions_apply
+
+	components := input.snapshot.components
+	some component in components
```

Comment:
If the application snapshot has 10 components, and two of them do not use a pinned reference, there will be 20 violations. Each component will report two violations. This is confusing.

Can we filter on components that match `input.image`? Maybe on the repo name?

Reply:
nice catch! Done